runners/spark/src/main/java/org/apache/beam/runners/spark/translation/BoundedDataset.java

```diff
@@ -98,9 +98,19 @@
   }
 
   @Override
-  public void cache(String storageLevel) {
-    // populate the rdd if needed
-    getRDD().persist(StorageLevel.fromString(storageLevel));
+  @SuppressWarnings("unchecked")
+  public void cache(String storageLevel, Coder<?> coder) {
+    StorageLevel level = StorageLevel.fromString(storageLevel);
+    if (level.equals(StorageLevel.MEMORY_ONLY())) {
```

Comment:
We should probably also check if the level is `StorageLevel.MEMORY_ONLY_2`

Reply:
right